kernel/aster-nix/src/time/timer.rs

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{
+    boxed::Box,
+    collections::BinaryHeap,
+    sync::{Arc, Weak},
+    vec::Vec,
+};
+use core::time::Duration;
+
+use aster_frame::{arch::register_timer_interrupt_callback, sync::SpinLock};
+
+use super::{clock::Clock, timer_callback::TimerCallback};
+
+/// A timer with periodic functionality.
+///
+/// Setting the timer will trigger a callback function upon expiration of
+/// the set time. If the timer has a interval time set, it will use
+/// the interval time to configure a new timing after expiration.
+pub struct IntervalTimer {
+    interval: SpinLock<Duration>,
+    timer_manager: Arc<TimerManager>,
+    registered_callback: Box<dyn Fn() + Send + Sync>,
+    timer_callback: SpinLock<Weak<TimerCallback>>,
+}
+
+impl IntervalTimer {
+    /// Create a `IntervalTimer` instance from a [`TimerManager`].
+    /// The generated [`TimerCallback`] of this timer will be managed by the `TimerManager`.
+    fn new<F>(registered_callback: F, timer_manager: Arc<TimerManager>) -> Arc<Self>
```

Comment:
Document to emphasize the `registered_callback` cannot sleep if it will eventually be executed in interrupt context.